Framework/DataObjects/src/Peak.cpp

```diff
@@ -466,8 +365,9 @@ double Peak::getDSpacing() const {
   // E_i,f=k_i,f^2*hbar^2/(2 m)
   return 1e10 * PhysicalConstants::h /
          sqrt(2.0 * PhysicalConstants::NeutronMass * PhysicalConstants::meV) /
-         sqrt(m_initialEnergy + m_finalEnergy -
```

Comment:
This needs to be reconciled with what is in `LeanPeak`. I'm suspicious the implementation can be moved into `BasePeak`, or we need to better document that `LeanPeak` assumes elastic scattering only.

Reply:
A "different" implementation would cache the energies so those values are only calculated once
```C++
  const auto initialEnergy = getInitialEnergy();
  const auto finalEnergy = getFinalEnergy();
  return 1e10 * PhysicalConstants::h /
         sqrt(2.0 * PhysicalConstants::NeutronMass * PhysicalConstants::meV) /
         sqrt(initialEnergy + finalEnergy -
              2.0 * sqrt(initialEnergy * finalEnergy) *
                  cos(two_theta));